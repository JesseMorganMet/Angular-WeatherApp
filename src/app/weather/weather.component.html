<div id="dailyCard">
  <li><img src="{{img}}" alt="{{alt}}"></li>
  <li>{{percip}}</li>
  <li>{{temp}}</li>
  <li>{{data.SiteRep.DV.Location.Period[i].value | date:'E'}}</li>
</div>

<!--<app-weather *ngFor="let day of days | async"></app-weather>-->
