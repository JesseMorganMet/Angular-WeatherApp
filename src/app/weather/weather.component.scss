li{
  list-style: none;
}
img{
  width:100%;
  max-width: 90px;
  margin: -10px 0 -10px 0;
}
#dailyCard{
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
