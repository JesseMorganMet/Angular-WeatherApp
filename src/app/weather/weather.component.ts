import {Component, Input, OnInit, Output} from '@angular/core';

@Component({
  selector: 'app-weather',
  templateUrl: './weather.component.html',
  styleUrls: ['./weather.component.scss']
})
export class WeatherComponent implements OnInit {

  @Input() data;
  @Input() i: number
  days
  temp
  percip
  wType
  img
  alt

  constructor() {
  }
  ngOnInit(): void {
    this.days = this.data.SiteRep.DV.Location.Period;
    this.avgTemp(this.days);
    this.avgPercip(this.days);
    this.weatherType(this.days);
  }

  avgTemp(days){
    const dayData = this.days[this.i].Rep; //hard coded for now
    let tempAvg = 0;
    for(let i =0; i < dayData.length; i++){
      let tempIndv = 0;
      tempIndv = parseInt(dayData[i].T);
      tempAvg = tempAvg + tempIndv;
    }
    this.temp = (tempAvg/dayData.length).toFixed(0) + "C";
    // console.log(temp)
  }

  avgPercip(days){
    const dayData = this.days[this.i].Rep; //hard coded for now
    let ppAvg = 0;
    for(let i =0; i < dayData.length; i++){
      let ppIndv = 0;
      ppIndv = parseInt(dayData[i].Pp);
      ppAvg = ppAvg + ppIndv;
    }
    let ppRound = Math.round((ppAvg/dayData.length)/5)*5;
    if(ppRound >= 10){
      ppRound = Math.round(ppRound/10)*10;
    }
    this.percip = (ppRound + "%");
    // console.log(percip)
  }
  weatherType(days){
    const dayData = this.days[this.i].Rep;
    let weatherDaily = parseInt(dayData[0].W);
    if (weatherDaily===1|| weatherDaily===0) {
      this.img = '../assets/icons/Sun.png';
      this.alt = 'Sunny Icon';
    } else if(weatherDaily===2||weatherDaily===3||weatherDaily===7||weatherDaily===8){
      this.img = '../assets/icons/Cloud.png';
      this.alt = 'Cloudy Icon';
    } else if(weatherDaily>=4&&weatherDaily<=6){
      this.img = '../assets/icons/Fog.png';
      this.alt = 'Foggy Icon';
    } else if(weatherDaily>=9&&weatherDaily<=15){
      this.img = '../assets/icons/Rain.png';
      this.alt = 'Raining Icon';
    } else if(weatherDaily>=16&&weatherDaily<=27){
      this.img = '../assets/icons/Snow.png';
      this.alt = 'Snowing Icon';
    } else if(weatherDaily>=28&&weatherDaily<=30){
      this.img = '../assets/icons/Thunder.png';
      this.alt = 'Thunder Icon';
    } else{
      console.log("Something went wrong");
      console.log(weatherDaily);
    }
  }
}
