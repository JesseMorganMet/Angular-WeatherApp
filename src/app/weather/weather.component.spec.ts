import { ComponentFixture, TestBed } from '@angular/core/testing';

import { WeatherComponent } from './weather.component';
import {AvgPrecipPipe, AvgTempPipe, WeatherTypeAltPipe, WeatherTypePipe} from '../avg-temp.pipe';

describe('WeatherComponent', () => {
  let component: WeatherComponent;
  let fixture: ComponentFixture<WeatherComponent>;

  const fakeData = {

        "value": "2022-08-15Z",
        "Rep": [
          {
            "Pp": "29",
            "T": "17",
            "W": "7",
            "U": "1",
          },
          {
            "Pp": "82",
            "T": "18",
            "W": "15",
            "U": "2",
          },
          {
            "Pp": "82",
            "T": "20",
            "W": "12",
            "U": "4",
          },
          {
            "Pp": "54",
            "T": "23",
            "W": "30",
            "U": "3",
          },
          {
            "Pp": "8",
            "T": "21",
            "W": "7",
            "U": "1",
          },
          {
            "Pp": "8",
            "T": "20",
            "W": "7",
            "U": "0",
          }
        ]
      }


  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [
        WeatherComponent,
        WeatherTypePipe,
        WeatherTypeAltPipe,
        AvgPrecipPipe,
        AvgTempPipe
      ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(WeatherComponent);
    component = fixture.componentInstance;
  });

  it('Should render the page correctly', () => {
    component.data = fakeData;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
