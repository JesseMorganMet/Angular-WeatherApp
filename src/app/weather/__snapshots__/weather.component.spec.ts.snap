// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherComponent Should render the page correctly 1`] = `
<app-weather
  __ngContext__={[Function LRootView]}
  data={[Function Object]}
>
  <div
    id="dailyCard"
  >
    <li>
      <img
        alt="Cloudy icon"
        src="../assets/icons/Cloud.png"
      />
    </li>
    <li>
      50%
    </li>
    <li>
      20C
    </li>
    <li>
      Mon
    </li>
  </div>
</app-weather>
`;
